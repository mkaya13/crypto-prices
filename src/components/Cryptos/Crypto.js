import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

export default function Crypto(props) {
  const { crypto } = props;

  return (
    <Link className="main-page-each-crypto" to={`/detail/${crypto && crypto.id}`}>
      <li>
        <div className="main-page-each-li-box">
          <img className="main-page-each-li-img" src={crypto && crypto.image} alt={crypto && crypto.image} />
          <div className="main-page-crypto-info">
            <h2 className="main-page-each-li-detail">
              <span className="detail-feature-tag">Name:</span>
              &nbsp;
              {crypto && crypto.name}
            </h2>
            <h2 className="main-page-each-li-detail">
              <span className="detail-feature-tag">Symbol:</span>
              &nbsp;
              {crypto && crypto.symbol.toUpperCase()}
            </h2>
            <h2 className="main-page-each-li-detail">
              <span className="detail-feature-tag">Price:</span>
              &nbsp;
              {crypto && crypto.currentPrice}
              &nbsp;
              <span className="USD">$</span>
            </h2>
          </div>
        </div>
      </li>
    </Link>
  );
}

Crypto.propTypes = {
  crypto: PropTypes.shape({
    id: PropTypes.string.isRequired,
    symbol: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    currentPrice: PropTypes.number.isRequired,
  }).isRequired,
};
