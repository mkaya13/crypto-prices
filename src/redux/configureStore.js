import { combineReducers, configureStore } from '@reduxjs/toolkit';
import { cryptosReducer } from './cryptos/cryptos';
import { commentsByIdReducer } from './comments/commentsById';

const rootReducer = combineReducers({
  cryptos: cryptosReducer,
  commentsById: commentsByIdReducer,
});

export const store = configureStore({
  reducer: rootReducer,
});
export default { store };
