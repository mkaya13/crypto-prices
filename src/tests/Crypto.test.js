import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';

import Crypto from '../components/Cryptos/Crypto';

import { store } from '../redux/configureStore';

it('renders Details correctly', () => {
  const tree = renderer.create(
    <BrowserRouter>
      <Provider store={store}><Crypto /></Provider>
    </BrowserRouter>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
