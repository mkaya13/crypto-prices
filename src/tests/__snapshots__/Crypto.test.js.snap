// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Crypto correctly 1`] = `
<a
  className="main-page-each-crypto"
  href="/detail/undefined"
  onClick={[Function]}
>
  <li>
    <div
      className="main-page-each-li-box"
    >
      <img
        className="main-page-each-li-img"
      />
      <div
        className="main-page-crypto-info"
      >
        <h2
          className="main-page-each-li-detail"
        >
          <span
            className="detail-feature-tag"
          >
            Name:
          </span>
           
        </h2>
        <h2
          className="main-page-each-li-detail"
        >
          <span
            className="detail-feature-tag"
          >
            Symbol:
          </span>
           
        </h2>
        <h2
          className="main-page-each-li-detail"
        >
          <span
            className="detail-feature-tag"
          >
            Price:
          </span>
           
           
          <span
            className="USD"
          >
            $
          </span>
        </h2>
      </div>
    </div>
  </li>
</a>
`;

exports[`renders Details correctly 1`] = `
<a
  className="main-page-each-crypto"
  href="/detail/undefined"
  onClick={[Function]}
>
  <li>
    <div
      className="main-page-each-li-box"
    >
      <img
        className="main-page-each-li-img"
      />
      <div
        className="main-page-crypto-info"
      >
        <h2
          className="main-page-each-li-detail"
        >
          <span
            className="detail-feature-tag"
          >
            Name:
          </span>
           
        </h2>
        <h2
          className="main-page-each-li-detail"
        >
          <span
            className="detail-feature-tag"
          >
            Symbol:
          </span>
           
        </h2>
        <h2
          className="main-page-each-li-detail"
        >
          <span
            className="detail-feature-tag"
          >
            Price:
          </span>
           
           
          <span
            className="USD"
          >
            $
          </span>
        </h2>
      </div>
    </div>
  </li>
</a>
`;
